### ISSUE TITLE
Move bit-set behind a cfg flag

### ISSUE THREAD

**Issue by fletcherw** [a1ien/rusb#29]
Would you be open to a pull request which converts bit-set to be an optional dependency based on a cfg option? If the cfg were disabled, DeviceHandle could just use the standard library's HashSet instead of BitSet

My reasoning is that for people who don't care about a speed or memory hit from using HashSet, there is no need to add the additional dependencies on bit-set and (transitively) bit-vec.

**Comment by a1ien** [comment]
I think we can remove BitSet and write simple wrapper for two `u128` instead. If you do this i am very happy to accept this PR.
For this we can add type 
``` rust

#[derive(Debug, Default, Copy, Clone)]
struct ClaimedInterface {
    inner: [u128;2];
}

impl ClaimedInterface {
  pub fn new() -> Self{
    Self {
    inner: [0,0],
    }
  }

 pub fn insert(&mut self, value: u8) {
...
 }
 
 pub fn remove(&mut self, value: usize) {
...
 }

}

```
And also implement iter. And implent two methods

**Comment by fletcherw** [comment]
Added a pull request to fix this. On a related note, when I was reading the libusb docs I saw the following in the comments for libusb_set_configuration "You cannot change/reset configuration if your application has claimed interfaces. It is advised to set the desired configuration before claiming interfaces.". Should we consider rejecting a call to set_configuration if we have claimed interfaces? Or is it better to just let libusb handle that internally?
